pcsx2/gui/AppCoreThread.cpp

```diff
@@ -427,23 +427,29 @@ static void _ApplySettings( const Pcsx2Config& src, Pcsx2Config& fixup )
 		// the BIos isn't loaded until after initial calls to ApplySettings.
 
 		gameName = L"Booting PS2 BIOS... ";
-	}
 
 	//Till the end of this function, entry CRC will be 00000000
-	if (!gameCRC.Length())
+	CDVD_SourceType cdvdsrc(g_Conf->CdvdSource);
+	if ((!gameCRC.Length()) && ((cdvdsrc != CDVD_SourceType::NoDisc)))
 	{
 		Console.WriteLn(Color_Gray, "Patches: No CRC found, using 00000000 instead.");
 		gameCRC = L"00000000";
 	}
+		else if ((!gameCRC.Length()) && ((cdvdsrc == CDVD_SourceType::NoDisc)))
+		gameCRC = L"00000000";
+	}
 
 	// regular cheat patches
-	if (fixup.EnableCheats)
+	CDVD_SourceType cdvdsrc(g_Conf->CdvdSource);
+	if ((fixup.EnableCheats) && (cdvdsrc != CDVD_SourceType::NoDisc))
+	{
 		gameCheats.Printf(L" [%d Cheats]", LoadPatchesFromDir(gameCRC, GetCheatsFolder(), L"Cheats"));
+	}
+		else if ((fixup.EnableCheats) && (cdvdsrc == CDVD_SourceType::NoDisc))
 
 	// wide screen patches
-	if (fixup.EnableWideScreenPatches)
-	{
-		if (int numberLoadedWideScreenPatches = LoadPatchesFromDir(gameCRC, GetCheatsWsFolder(), L"Widescreen hacks"))
+	if ((fixup.EnableWideScreenPatches) && (cdvdsrc == CDVD_SourceType::NoDisc));
+		else if (int numberLoadedWideScreenPatches = LoadPatchesFromDir(gameCRC, GetCheatsWsFolder(), L"Widescreen hacks"))
```

Comment:
Yeah I knew there was a better way than doing an empty check, could t think of it at the time.